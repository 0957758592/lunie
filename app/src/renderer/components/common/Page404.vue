<template lang="pug">
page(title='404 - Page Not Found')
  part(title='Try another page...')
    tm-list-item(to='/' title='Balances')
    tm-list-item(to='/staking' title='Staking')
</template>

<script>
import { TmListItem } from "@tendermint/ui"
import Page from "common/NiPage"
import Part from "common/NiPart"
export default {
  name: "page-404",
  components: {
    TmListItem,
    Page,
    Part
  }
}
</script>
