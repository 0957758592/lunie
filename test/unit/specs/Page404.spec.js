import { mount } from 'vue-test-utils'
import Page404 from 'renderer/components/common/Page404'

describe('Page404', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(Page404, {
      propsData: { }
    })
  })

  it('has the expected html structure', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it('should show links to other pages', () => {
    expect(wrapper.findAll('.ni-li').length > 0).toBe(true)
  })
})
