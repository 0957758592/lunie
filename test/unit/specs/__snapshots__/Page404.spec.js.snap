// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404 has the expected html structure 1`] = `
<div
  class="ni-page"
>
  <header
    class="ni-page-header"
  >
    <div
      class="ni-page-header-container"
    >
      <div
        class="ni-page-header-text"
      >
        <div
          class="ni-page-header-title"
        >
          <div>
            404 - Page Not Found
          </div>
        </div>
        <div
          class="ni-page-header-subtitle"
        >
          <div>
            
          </div>
        </div>
      </div>
      <menu
        class="ni-page-header-menu"
      >
        <div />
      </menu>
    </div>
  </header>
  <main
    class="ni-page-main"
  >
    <section
      class="ni-part"
    >
      <div
        class="ni-part-container"
      >
        <header
          class="ni-part-header"
        >
          <div
            class="ni-part-header-container"
          >
            <div
              class="ni-part-title h5"
            >
              Try another page...
            </div>
            <menu
              class="ni-part-menu"
            />
          </div>
        </header>
        <main
          class="ni-part-main"
        >
          <router-link
            class="ni-li ni-li-link"
            to="/"
          >
            <div
              class="ni-li-container"
            >
              <div
                class="ni-li-thumb"
              >
                <!---->
              </div>
              <div
                class="ni-li-label"
              >
                <div
                  class="ni-li-title"
                >
                  Balance
                </div>
                <div
                  class="ni-li-subtitle"
                >
                  
                </div>
              </div>
              <div
                class="ni-li-icon"
              >
                <i
                  class="material-icons inactive"
                >
                  chevron_right
                </i>
                <i
                  class="material-icons active"
                >
                  my_location
                </i>
              </div>
            </div>
          </router-link>
          <router-link
            class="ni-li ni-li-link"
            to="/staking"
          >
            <div
              class="ni-li-container"
            >
              <div
                class="ni-li-thumb"
              >
                <!---->
              </div>
              <div
                class="ni-li-label"
              >
                <div
                  class="ni-li-title"
                >
                  Staking
                </div>
                <div
                  class="ni-li-subtitle"
                >
                  
                </div>
              </div>
              <div
                class="ni-li-icon"
              >
                <i
                  class="material-icons inactive"
                >
                  chevron_right
                </i>
                <i
                  class="material-icons active"
                >
                  my_location
                </i>
              </div>
            </div>
          </router-link>
        </main>
      </div>
    </section>
  </main>
</div>
`;
